cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/AbstractCompositeTrigger.java

```diff
@@ -85,4 +88,14 @@ private void initializeUnitTriggers() {
       }
     }
   }
+
+  public List<TriggerInfo> getUnitTriggerInfos(TriggerInfoContext context) {
```

Comment:
Yes, each composite trigger is like a node in the tree, and unit triggers are like the leaf nodes in the tree. The tree structure is not stored in the map, but is represented by the composite trigger which contains other triggers. This `getUnitTriggerInfos` method just need to return all the unique leaf nodes in the tree, i.e. unique unit triggers.